### PR TITLE
Add S3 Object resource 

### PR DESCRIPTION
**Pull request by zbbkeepgoing**
Signed-off-by: Binbin Zou <zbbkeepgoing@163.com>

### Description of your changes

This PR adds an Object resource for S3 bucket that allows create object or directory in a bucket using Crossplane. 

I have:

- [x] Read and followed Crossplane's [contribution process].
- [x] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested
Manually and through unit tests

[contribution process]: https://git.io/fj2m9

### CODE REVIEW

**Comment by zbbkeepgoing** [comment]
Can anyone review this PR？

**Comment by haarchri** [comment]
@zbbkeepgoing can you add an example for s3 object?

**Comment by zbbkeepgoing** [comment]
> @zbbkeepgoing can you add an example for s3 object?

nice advice, done

**Review by haarchri — APPROVED** [review]

**Comment by zbbkeepgoing** [comment]
@haarchri Hi, is this pr can be merged?

File: examples/s3/object.yaml [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    crossplane.io/external-name: test.txt
+spec:
+  deletionPolicy: Delete
+  providerConfigRef:
+    name: default
+  forProvider:
+    region: us-west-2
+    key: path/test.txt
+    acl: private
+    body: "this is test.txt"
```
Comment (haarchri) [inline examples/s3/object.yaml:32]: guess its better to have input for body/key out of ConfigMaps/Secrets ?

**Pull request closed without merging**